### PR TITLE
docs: Improved README & documentation

### PR DESCRIPTION
**Pull request by fg-mindee**
This PR introduces the following modifications:
- rewrites the README completely
- added explanations about how we model documents

_Note to reviewer: I suggest checking the README rendering on the branch directly_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,88 +3,104 @@
 
 [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE) ![Build Status](https://github.com/mindee/doctr/workflows/python-package/badge.svg) [![codecov](https://codecov.io/gh/mindee/doctr/branch/main/graph/badge.svg?token=577MO567NM)](https://codecov.io/gh/mindee/doctr) [![CodeFactor](https://www.codefactor.io/repository/github/mindee/doctr/badge?s=bae07db86bb079ce9d6542315b8c6e70fa708a7e)](https://www.codefactor.io/repository/github/mindee/doctr) [![Codacy Badge](https://api.codacy.com/project/badge/Grade/340a76749b634586a498e1c0ab998f08)](https://app.codacy.com/gh/mindee/doctr?utm_source=github.com&utm_medium=referral&utm_content=mindee/doctr&utm_campaign=Badge_Grade) [![Doc Status](https://github.com/mindee/doctr/workflows/doc-status/badge.svg)](https://mindee.github.io/doctr) [![Pypi](https://img.shields.io/badge/pypi-v0.1.1-blue.svg)](https://pypi.org/project/python-doctr/) 
 
-Extract valuable information from your documents.
 
+**Optical Character Recognition made seamless & accessible to anyone, powered by TensorFlow 2.0**
```
Comment (charlesmindee) [inline README.md:7]: maybe tensorflow 2 instead of 2.0 since we are using tf 2.4 ?

File: README.md [hunk 2]
```diff
@@ -3,88 +3,104 @@
...
+
+
+## Models architectures
+Credits where it's due: this repository is implementing, among others, architectures from published research papers.
+
+### Text Detection
+- [Real-time Scene Text Detection with Differentiable Binarization](https://arxiv.org/pdf/1911.08947.pdf).
+
+### Text Recognition
+- [Convolutional RNN: an Enhanced Model for Extracting Features from Sequential Data](https://arxiv.org/pdf/1602.05875.pdf).
```
Comment (charlesmindee) [inline README.md:95]: The paper for the CRNN is the following: https://arxiv.org/abs/1507.05717 (An End-to-End Trainable Neural Network for Image-based Sequence Recognition and Its Application to Scene Text Recognition)

**Review by charlesmindee — CHANGES_REQUESTED** [review]
Thanks for the PR! I have only 2 suggestions

File: README.md (hunk 2 above)
Comment (fg-mindee) [reply README.md]: Good to know, I just fixed the other reference to the paper!

**Review by charlesmindee — APPROVED** [review]

**Pull request merged**